### PR TITLE
SOLR-16132: Stop gpg-agent after use

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16132

### CODE REVIEW

File: dev-tools/scripts/smokeTestRelease.py [hunk 1]
```diff
@@ -277,16 +277,19 @@ def checkSigs(urlString, version, tmpDir, isSigned, keysFile):
   if os.path.exists(gpgHomeDir):
     shutil.rmtree(gpgHomeDir)
   os.makedirs(gpgHomeDir, 0o700)
-  run('gpg --homedir %s --import %s' % (gpgHomeDir, keysFile),
-      '%s/solr.gpg.import.log' % tmpDir)
+  gpgLogFile = '%s/solr.gpg.import.log' % tmpDir
+  run('gpg --homedir %s --import %s' % (gpgHomeDir, keysFile), gpgLogFile)
```
Comment (risdenk) [inline dev-tools/scripts/smokeTestRelease.py:281]: This GPG stuff would be nicer to move after testChanges? or maybe in `isSigned`? If we aren't checking `isSigned` do we care about this gpg import?

File: dev-tools/scripts/smokeTestRelease.py [hunk 2]
```diff
@@ -324,6 +327,11 @@ def checkSigs(urlString, version, tmpDir, isSigned, keysFile):
           if line.lower().find('warning') != -1:
             print('      GPG: %s' % line.strip())
 
+      # Make sure to shutdown the GPG agent at the end
```
Comment (risdenk) [inline dev-tools/scripts/smokeTestRelease.py:330]: I think line 323 above is missing `--homedir`...

**Comment by risdenk** [comment]
Well this fixed the explicit gpg invocations.

It did NOT fix the Maven check it looks like:

```
...
verify maven artifact sigs
...
```

I saw the gpg agent get created and shutdown at the beginning. I need to see where Maven is calling this to fix it too:

```
risdenk          42369   0.0  0.0 408637808   2256   ??  S    12:38PM   0:00.00 scdaemon --multi-server --homedir /tmp/smoke_solr_9.1.0_aa4f3d98ab19c201e7f3c74cd14c99174148616d_3/solr.gpg
risdenk          42368   0.0  0.0 408657760   1472   ??  Ss   12:38PM   0:00.01 gpg-agent --homedir /tmp/smoke_solr_9.1.0_aa4f3d98ab19c201e7f3c74cd14c99174148616d_3/solr.gpg --use-standard-socket --daemon
```

**Comment by risdenk** [comment]
This works after the verifyMavenSigs as well now. No more left over gpg agent :)

**Review by HoustonPutman — APPROVED** [review]

**Pull request merged**